### ISSUE TITLE
[java] LocalVariableCouldBeFinal false positive with try/catch

### ISSUE THREAD

**Issue by acanda** [pmd/pmd#5046]
**Affects PMD Version:** 
7.2.0

**Rule:**
[LocalVariableCouldBeFinal](https://docs.pmd-code.org/pmd-doc-7.2.0/pmd_rules_java_codestyle.html#localvariablecouldbefinal)

**Code Sample demonstrating the issue:**

```java
    private Optional<FileWatcher> createFileWatcher() {
        @SuppressWarnings("PMD.LocalVariableCouldBeFinal")
        Optional<FileWatcher> optionalFileWatcher;  // false positive in PMD 7.2.0, cannot be final
        try {
            optionalFileWatcher = Optional.of(new FileWatcher());
        } catch (final IOException e) {
            optionalFileWatcher = Optional.empty();
        }
        return optionalFileWatcher;
    }

```

**Expected outcome:**
This is new in 7.2.0 and worked in prior versions.
PMD reports a violation, but that's wrong. That's a false positive.

**Running PMD through:**
Maven

**Comment by c0va23** [comment]
We faced wit same issue.

**Comment by zman0900** [comment]
Remains a problem with 7.3.0

**Comment by rbri** [comment]
Same here with 7.3.0 - any hope to get this fixed?

**Comment by oowekyala** [comment]
I will take a look...

**Comment by zman0900** [comment]
Happens with 7.4.0 also

**Comment by Wolf2323** [comment]
Still an issue with 7.5.0

**Comment by rbri** [comment]
Thanks for taking care of this.

**Comment by uckelman** [comment]
Appears fixed in 7.7.0.
